.github/workflows/backwards_compatibility_tests_workflow.yml

```diff
@@ -31,12 +32,43 @@ jobs:
         with:
           java-version: ${{ matrix.java }}
 
-      - name: Install dependencies
+      - name: Install dependencies on ubuntu
+        if: startsWith(matrix.os,'ubuntu')
         run: |
           sudo apt-get install libopenblas-dev gfortran -y
 
+      - name: Install MinGW Using Scoop
+        if: startsWith(matrix.os,'windows')
+        run: |
+          iex "& {$(irm get.scoop.sh)} -RunAsAdmin"
+          scoop bucket add main
+          scoop install mingw
+
+      - name: Add MinGW to PATH
+        if: startsWith(matrix.os,'windows')
+        run: |
+          echo "C:/Users/runneradmin/scoop/apps/mingw/current/bin" >> $env:GITHUB_PATH
+          refreshenv
+
+      - name: Download OpenBLAS
+        if: startsWith(matrix.os,'windows')
+        run: |
+          curl -L -O https://github.com/xianyi/OpenBLAS/releases/download/v0.3.21/OpenBLAS-0.3.21-x64.zip
+          mkdir OpenBLAS
+          Expand-Archive -Path .\OpenBLAS-0.3.21-x64.zip -DestinationPath .\OpenBLAS\
+          mkdir ./src/main/resources/windowsDependencies
+          cp ./OpenBLAS/bin/libopenblas.dll ./src/main/resources/windowsDependencies/
+          rm .\OpenBLAS-0.3.21-x64.zip
+          rm -r .\OpenBLAS\
 
-      - name: Run k-NN Restart-Upgrade BWC Tests from BWCVersion-${{ matrix.bwc_version }} to OpenSearch Version-${{ matrix.opensearch_version }}
+      - name: Run k-NN Restart-Upgrade BWC Tests from BWCVersion-${{ matrix.bwc_version }} to OpenSearch Version-${{ matrix.opensearch_version }} Windows
+        if: (startsWith(matrix.os,'windows') && (startsWith(matrix.bwc_version, '1.3') || startsWith(matrix.bwc_version, '2.4') || startsWith(matrix.bwc_version, '2.5') || startsWith(matrix.bwc_version, '2.6') || startsWith(matrix.bwc_version, '3.')))
+        run: |
+          echo "Running restart-upgrade backwards compatibility tests ..."
+          ./gradlew :qa:restart-upgrade:testRestartUpgrade -D'tests.bwc.version=${{ matrix.bwc_version }}'
+
+      - name: Run k-NN Restart-Upgrade BWC Tests from BWCVersion-${{ matrix.bwc_version }} to OpenSearch Version-${{ matrix.opensearch_version }} Ubuntu
```

Comment:
Can we combine this with previous task?

Reply:
It is not supporting to include both if conditions in one step. It doesn't support if/else as well